tests/test_modin.py

```diff
@@ -11,12 +11,13 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-
+import pytest
 
 from nums.core.array.blockarray import BlockArray
 
 
 # pylint: disable=import-outside-toplevel
+@pytest.mark.timeout(30)
```

Comment:
Will this be temporary once we find the bug? Seems like it's useful for debugging purposes but comes along with consequences. https://pypi.org/project/pytest-timeout/

If possible, you could try to rerun the workflows with a long for loop over `test_modin()`. Then see what the stack trace is if it happens to catch on a hang/timeout.